mypy/checkpattern.py

```diff
@@ -205,12 +204,15 @@ def visit_value_pattern(self, o: ValuePattern) -> PatternType:
         current_type = self.type_context[-1]
         typ = self.chk.expr_checker.accept(o.expr)
         typ = coerce_to_literal(typ)
```

Comment:
Is this `coerce_to_literal` still necessary now that we use `narrow_type_by_equality` rather than `conditional_types_with_intersection`?

Reply:
It probably is; we are not in literal context, so coercion won't happen there for `==`. Removing this will change the behavior significantly.